src/cephadm/cephadm

```diff
@@ -1943,13 +1948,19 @@ def infer_fsid(func: FuncT) -> FuncT:
 
 def infer_config(func: FuncT) -> FuncT:
     """
-    If we find a MON daemon, use the config from that container
+    Infer the clusater configuration using the followign priority order:
+     1- if the user has provided custom conf file (-c option) use it
+     2- otherwise if daemon --name has been provided use daemon conf
+     3- otherwise if {ctx.data_dir}/{fsid}/{CEPH_CONF_DIR} dir exists use it
+     4- otherwise find the mon daemon conf file and use it
```

Comment:
3 and 4 should be swapped imo. If we have a mon on the host still use that (which preserves previous behavior) then we have this new conf location to possibly use over /etc/ceph/

Reply:
Fixed, now we first look for the mon daemon conf file